core/src/main/scala/kafka/admin/ConfigCommand.scala

```diff
@@ -533,65 +522,120 @@ object ConfigCommand extends Config {
             .ofType(classOf[String])
             .withValuesSeparatedBy(',')
     val forceOpt = parser.accepts("force", "Suppress console prompts")
+    val topic = parser.accepts("topic", "The topic's name.")
+      .withRequiredArg
+      .ofType(classOf[String])
+    val client = parser.accepts("client", "The client's ID.")
+      .withRequiredArg
+      .ofType(classOf[String])
+    val clientDefaults = parser.accepts("client-defaults", "The config defaults for all clients.")
+    val user = parser.accepts("user", "The user's principal name.")
```

Comment:
`kafka-acls.sh` has a `--principal` flag, which I think is how we really should be referring to this.  Principals aren't really "users," although we sometimes refer to them that way informally.  Also, it's important to keep consistency between tools.

It's unfortunate that `kafka-configs.sh` already accepts `--entity-type users`.  I think it would be better to deprecate that, and migrate to `--entity-type principals` as the recommended usage.  Thoughts?